# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.h
# Commit: de69e3c1635a
# Full Hash: de69e3c1635a311b5b22716b65b93fc1899a37a4
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-08-01 09:48:10
# Regressor Bug: 1613431
# File Overlap Count: 1
# Description:
#   Bug 1613431 - Part 3: Handle synced setters return value. r=nika
#   
#   Depends on D83646
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83647
# ==============================================================================

diff -r b326d33f98cb -r de69e3c1635a docshell/base/BrowsingContext.h
--- a/docshell/base/BrowsingContext.h	Fri Jul 31 13:37:00 2020 +0000
+++ b/docshell/base/BrowsingContext.h	Fri Jul 31 13:37:13 2020 +0000
@@ -338,7 +338,8 @@
   void SetOpener(BrowsingContext* aOpener) {
     MOZ_DIAGNOSTIC_ASSERT(!aOpener || aOpener->Group() == Group());
     MOZ_DIAGNOSTIC_ASSERT(!aOpener || aOpener->mType == mType);
-    SetOpenerId(aOpener ? aOpener->Id() : 0);
+
+    MOZ_ALWAYS_SUCCEEDS(SetOpenerId(aOpener ? aOpener->Id() : 0));
   }
 
   bool HasOpener() const;
@@ -356,12 +357,15 @@
   already_AddRefed<BrowsingContext> GetOnePermittedSandboxedNavigator() const {
     return Get(GetOnePermittedSandboxedNavigatorId());
   }
-  void SetOnePermittedSandboxedNavigator(BrowsingContext* aNavigator) {
+  MOZ_MUST_USE nsresult
+  SetOnePermittedSandboxedNavigator(BrowsingContext* aNavigator) {
     if (GetOnePermittedSandboxedNavigatorId()) {
       MOZ_ASSERT(false,
                  "One Permitted Sandboxed Navigator should only be set once.");
+      return NS_ERROR_FAILURE;
     } else {
-      SetOnePermittedSandboxedNavigatorId(aNavigator ? aNavigator->Id() : 0);
+      return SetOnePermittedSandboxedNavigatorId(aNavigator ? aNavigator->Id()
+                                                            : 0);
     }
   }
 
@@ -398,7 +402,8 @@
   void SetUsePrivateBrowsing(bool aUsePrivateBrowsing, ErrorResult& aError);
   // Needs a different name to disambiguate from the xpidl method with
   // the same signature but different return value.
-  void SetUseTrackingProtectionWebIDL(bool aUseTrackingProtection);
+  void SetUseTrackingProtectionWebIDL(bool aUseTrackingProtection,
+                                      ErrorResult& aRv);
   bool UseTrackingProtectionWebIDL() { return UseTrackingProtection(); }
   void GetOriginAttributes(JSContext* aCx, JS::MutableHandle<JS::Value> aVal,
                            ErrorResult& aError);
@@ -434,24 +439,30 @@
   }
 
   // ScreenOrientation related APIs
-  void SetCurrentOrientation(OrientationType aType, float aAngle) {
-    SetCurrentOrientationType(aType);
-    SetCurrentOrientationAngle(aAngle);
+  MOZ_MUST_USE nsresult SetCurrentOrientation(OrientationType aType,
+                                              float aAngle) {
+    Transaction txn;
+    txn.SetCurrentOrientationType(aType);
+    txn.SetCurrentOrientationAngle(aAngle);
+    return txn.Commit(this);
   }
 
-  void SetRDMPaneOrientation(OrientationType aType, float aAngle) {
+  void SetRDMPaneOrientation(OrientationType aType, float aAngle,
+                             ErrorResult& aRv) {
     if (InRDMPane()) {
-      SetCurrentOrientation(aType, aAngle);
+      if (NS_FAILED(SetCurrentOrientation(aType, aAngle))) {
+        aRv.ThrowInvalidStateError("Browsing context is discarded");
+      }
     }
   }
 
-  void SetRDMPaneMaxTouchPoints(uint8_t aMaxTouchPoints) {
+  void SetRDMPaneMaxTouchPoints(uint8_t aMaxTouchPoints, ErrorResult& aRv) {
     if (InRDMPane()) {
-      SetMaxTouchPointsOverride(aMaxTouchPoints);
+      SetMaxTouchPointsOverride(aMaxTouchPoints, aRv);
     }
   }
 
-  void SetAllowContentRetargeting(bool aAllowContentRetargeting);
+  nsresult SetAllowContentRetargeting(bool aAllowContentRetargeting);
 
   // Using the rules for choosing a browsing context we try to find
   // the browsing context with the given name in the set of
@@ -551,12 +562,13 @@
   void GetCustomUserAgent(nsAString& aUserAgent) {
     aUserAgent = Top()->GetUserAgentOverride();
   }
-  void SetCustomUserAgent(const nsAString& aUserAgent);
+  nsresult SetCustomUserAgent(const nsAString& aUserAgent);
+  void SetCustomUserAgent(const nsAString& aUserAgent, ErrorResult& aRv);
 
   void GetCustomPlatform(nsAString& aPlatform) {
     aPlatform = Top()->GetPlatformOverride();
   }
-  void SetCustomPlatform(const nsAString& aPlatform);
+  void SetCustomPlatform(const nsAString& aPlatform, ErrorResult& aRv);
 
   JSObject* WrapObject(JSContext* aCx);
 
@@ -568,7 +580,7 @@
 
   void StartDelayedAutoplayMediaComponents();
 
-  void ResetGVAutoplayRequestStatus();
+  MOZ_MUST_USE nsresult ResetGVAutoplayRequestStatus();
 
   /**
    * Information required to initialize a BrowsingContext in another process.